### PR TITLE
fix(checkout): CHECKOUT-4774 Handle no hosted fields to be rendered scenario

### PR DESCRIPTION
**Pull request by ankurbigcomm**
## What?
Handle scenario where hosted fields are not available to be rendered.

## Why?
If a merchant has configured vaulted cards with vaulted cvv code since there is no field expected to be filled out, the hosted form should not be rendered.

## Testing Proof
Refer PR https://github.com/bigcommerce/checkout-js/pull/261

@bigcommerce/checkout @bigcommerce/payments

### CODE REVIEW

File: src/payment/strategies/credit-card/credit-card-payment-strategy.ts [hunk 1]
```diff
@@ -98,4 +104,8 @@ export default class CreditCardPaymentStrategy implements PaymentStrategy {
 
         return isHostedPaymentFormEnabled;
     }
+
+    private _ishostedFieldAvailable(options?: PaymentInitializeOptions): boolean {
```
Comment (davidchin) [inline src/payment/strategies/credit-card/credit-card-payment-strategy.ts:108]: Can you capitalise the `h`? 💅

File: src/payment/strategies/credit-card/credit-card-payment-strategy.ts [hunk 2]
```diff
@@ -98,4 +104,8 @@ export default class CreditCardPaymentStrategy implements PaymentStrategy {
...
+        return !(values(options && options.creditCard && options.creditCard.form.fields).every(isUndefined));
```
Comment (davidchin) [inline src/payment/strategies/credit-card/credit-card-payment-strategy.ts:109]: Should probably consider `null` as well. Perhaps `isNil` is more suitable as it covers both `undefined` and `null`.

File: src/payment/strategies/credit-card/credit-card-payment-strategy.spec.ts [hunk 3]
```diff
@@ -196,4 +196,48 @@ describe('CreditCardPaymentStrategy', () => {
...
+                ));
+
+            jest.spyOn(orderActionCreator, 'loadCurrentOrder')
+                .mockReturnValue(loadOrderAction);
+
+            jest.spyOn(formFactory, 'create')
+                .mockReturnValue(form);
+        });
+
+        it('does not create hosted form', async () => {
```
Comment (davidchin) [inline src/payment/strategies/credit-card/credit-card-payment-strategy.spec.ts:236]: Can you also add a case to test that we're not submitting using the hosted form in this scenario?

Comment (ankurbigcomm) [reply src/payment/strategies/credit-card/credit-card-payment-strategy.spec.ts:236]: Done

**Pull request merged**